website/conferences/views.py

```diff
@@ -180,14 +184,122 @@ def _render_conference_node(node, idx, conf):
+                    INNER JOIN "osf_contributor" U1 ON (U0."id" = U1."user_id")
+                    WHERE (U1."node_id" = ("osf_abstractnode"."id")
+                        AND U1."visible" = true)
+                    ORDER BY U1."_order" ASC
+                    LIMIT 1)::text) LIKE UPPER(%s))
+            """, [format_q, format_q]
+        )
+        # Turns the raw queryset back into a queryset - we still need to sort and paginate it.
+        nodes = nodes.filter(id__in=(x.id for x in raw_queryset))
+
+    if 'title' in sort or 'created' in sort:
+        nodes = nodes.order_by(sort)
+    elif 'author' in sort:
+        nodes = nodes.extra(select={
+            'author': '\
+                (SELECT U0."family_name" \
+                 FROM "osf_osfuser" U0 \
+                 INNER JOIN "osf_contributor" U1 ON (U0."id" = U1."user_id") \
+                 WHERE (U1."node_id" = ("osf_abstractnode"."id") AND U1."visible" = true) \
+                 ORDER BY U1."_order" ASC LIMIT 1)'
+        })
+        nodes = nodes.extra(order_by=[sort])
+    elif 'category' in sort:
+        tag_subqs = Tag.objects.filter(
+            abstractnode_tagged=OuterRef('pk'),
+            name=conf.field_names['submission1']).values_list('name', flat=True)
+        nodes = nodes.annotate(sub_one_count=Count(Subquery(tag_subqs))).annotate(
+            sub_name=Case(
+                When(sub_one_count=1, then=Value(conf.field_names['submission1'])),
+                default=Value(conf.field_names['submission2']),
+                output_field=CharField()
+            )
+        ).order_by('-sub_name' if '-' in sort else 'sub_name')
+    elif 'downloads' in sort:
+        pages = PageCounter.objects.annotate(
+            node_id=Substr('_id', 10, 5),
+            file_id=Substr('_id', 16),
+            _id_length=Length('_id')
+        ).filter(
+            _id__icontains='download',
+            node_id=OuterRef('guids___id'),
+            file_id=OuterRef('file_id')
+        ).exclude(_id_length__gt=39)
+        file_subqs = OsfStorageFile.objects.filter(node=OuterRef('pk')).order_by('created')
+        nodes = nodes.annotate(
+            file_id=Subquery(file_subqs.values('_id')[:1])
+        ).annotate(
+            downloads=Coalesce(Subquery(pages.values('total')[:1]), Value(0))
+        ).order_by(sort)
+    return nodes
+
+def paginated_conference_data(meeting, conf, meetings_page=1):
+    """
+    Returns a paginated, sorted, filtered array of conference nodes
+    :param str meeting: Endpoint name for a conference.
+    :param object conf: Conference
+    :param int meetings_page: Requested page of results, 1 by default
+    """
+    nodes = filter_and_sort_conference_data(AbstractNode.objects.filter(
+        tags__id__in=Tag.objects.filter(
+            name__iexact=meeting, system=False
+        ).values_list('id', flat=True), is_public=True, is_deleted=False), conf)
+
+    paginator = Paginator(nodes, SUBMISSIONS_PER_PAGE)
+    try:
```

Comment:
This was a good point; I thought about this and prefer sticking with [Django's invalid page exceptions](https://docs.djangoproject.com/en/2.0/topics/pagination/#invalidpage-exceptions). However I got rid of the `get_meetings_page` method you commented on below, because I could also use the invalid page exceptions for this. I think there's a small advantage in using their already-built exceptions rather than duplicating their logic to avoid throwing those same exceptions.